packages/gestalt/src/AbstractEvent.js

```diff
@@ -0,0 +1,14 @@
+// @flow strict
+
+/**
+ * We use a common event payload across all components. Since it's different from standard events that people are used to from React, it's abstracted here to at least make it standard. If you have to learn something, you should only have to learn it once.
+ *
+ * Why is it non-standard? Gestalt components are controlled - meaning that the can contain state and when that changes it's reflected back to the parent. These state changes are propagated via. event handlers. Often this state doesn't represent what's in the DOM or we just want to abstract the DOM from the consumer. We felt this format struct a good balance between readability and prediability.
+ */
+export type AbstractEvent<
+  T: SyntheticEvent<HTMLElement> | Event,
+  U = {||}
+> = ({|
```

Comment:
I know this is compatible with the existing API, but I am curious why we decided to go with the current approach.

Have we considered the following API where the first parameter is the event object and the second parameter is whatever additional data we want to return?

```
type AbstractEventHandler<T: SyntheticEvent<HTMLElement> | Event, U>
= (T, U) => void;
```

There are several benefits of this approach:

- This is more consistent with React's built-in event handler API where the first parameter is event object. 
- The second argument doesn't have to be an object
- There is no implicit requirement that the payload object cannot contain property named "event"

What do you think?

Reply:
I think you should either make an issue or open a PR and we can discuss further :) The intention of this PR wasn't to change the existing signature but to make it easier to change across the board in the future.

Reply:
sounds good. thanks!